### PR TITLE
Fixed issue #19203: alphabetical order didn't respect survey language

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: use and send QuestionBaseRenderer->sLanguage
Dev: follow it until needed in Question

### CODE REVIEW

File: application/models/Question.php [hunk 1]
```diff
@@ -774,15 +781,17 @@ private function sortAnswerOptions($answerOptions)
 
         // Sort alphabetically if applicable
         if ($this->shouldOrderAnswersAlphabetically()) {
+            if (is_null($language)) {
```
Comment (Shnoulle) [inline application/models/Question.php:784]: did we need to check if language exist in survey->getAllLanguade ? Think it's already done in QuestionBaseRenderer ?

File: application/models/Question.php [hunk 2]
```diff
@@ -774,15 +781,17 @@ private function sortAnswerOptions($answerOptions)
...
+                $language = $this->survey->language;
+            }
             foreach ($answerOptions as $scaleId => $scaleArray) {
                 $sorted = array();
 
-                // We create an aray sorted that will use the answer in the current language as key, and that will store its old index as value
+                // We create an array sorted that will use the answer in the current language as key, and that will store its old index as value
                 foreach ($scaleArray as $key => $answer) {
-                    $sorted[$answer->answerl10ns[$this->survey->language]->answer] = $key;
+                    $sorted[$answer->answerl10ns[$language]->answer] = $key;
```
Comment (Shnoulle) [inline application/models/Question.php:792]: Check if isset ?

I can happen after import TSV or have specific issue …

File: application/models/Question.php (hunk 1 above)
Comment (gabrieljenik) [reply application/models/Question.php:784]: Maybe double checking here wouldn't be the worst as to make things more atomic.
Just checking if the first answer is set would be enough, right?

File: application/models/Question.php (hunk 2 above)
Comment (gabrieljenik) [reply application/models/Question.php:792]: I think this s covered on the comments above.

**Review by gabrieljenik — APPROVED** [review]
Code structure looks OK.

File: application/models/Question.php (hunk 1 above)
Comment (Shnoulle) [reply application/models/Question.php:784]: Ok

File: application/models/Question.php (hunk 2 above)
Comment (Shnoulle) [reply application/models/Question.php:792]: > I think this s covered on the comments above.

Not : _I can happen after import TSV or have specific issue …_ 

If you can not do the database repair : it happen a lot … but no issue with debug = 0

Comment (gabrieljenik) [reply application/models/Question.php:792]: Sure. Let's check also then. It doesn't hurt.

Comment (Shnoulle) [reply application/models/Question.php:792]: I check : seems OK with broken language to

Comment (Shnoulle) [reply application/models/Question.php:792]: It break before

File: application/models/Question.php [hunk 3]
```diff
@@ -781,7 +781,7 @@ private function sortAnswerOptions($answerOptions, $language = null)
 
         // Sort alphabetically if applicable
         if ($this->shouldOrderAnswersAlphabetically()) {
-            if (is_null($language)) {
+            if (empty($language) || !in_array($language, $this->survey->allLanguages)) {
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/models/Question.php:784]: $language used ONLY for Alphabetically order.

The i'm sure it muts be tested only of needed

**Pull request merged**